Remove `OTP_SECRET` env configuration

mjankowski opened the pull request:
Follow-up on https://github.com/mastodon/mastodon/pull/34743

mjankowski committed:
Remove `OTP_SECRET` env configuration

ClearlyClaire commented:
> In added commit I gave this a placeholder value (the previous test env value) because (I think?) the assumption is that this migration will only ever be run as part of these tests in the future, and the data in the test was generated as though this were the value. Open to other ideas here, and for double-check of my assumption.

This will break migrations from earlier versions when skipping releases (which iirc we are still supporting, just not with zero-downtime).

ClearlyClaire commented:
> We probably can remove the setup script stuff, though? Since new installs at least no longer need the env var.

Definitely, yes

mjankowski commented:
Rebased after https://github.com/mastodon/mastodon/pull/34757 and also reverted the migration back to read from the config, along with restoring the config in production (for version skipping upgrades) and test env (for migration checks).

This leaves removal from the User model and dev env. We should maybe also change the TODO note around some of the other spots to better describe why its still needed, and/or conditions to actually remove (4.3 being EOL?)
